src/main/java/github/scarsz/discordsrv/listeners/PlayerAchievementsListener.java

```diff
@@ -49,18 +53,41 @@ public void onPlayerAchievementAwarded(PlayerAchievementAwardedEvent event) {
         if (PlayerUtil.isVanished(event.getPlayer())) return;
 
         // turn "ACHIEVEMENT_NAME" into "Achievement Name"
+        TextChannel channel = DiscordSRV.getPlugin().getMainTextChannel();
+        String channelName = channel.getName();
         String achievementName = PrettyUtil.beautify(event.getAchievement());
+        String message = LangUtil.Message.PLAYER_ACHIEVEMENT.toString();
+        Player player = event.getPlayer();
 
-        String discordMessage = LangUtil.Message.PLAYER_ACHIEVEMENT.toString()
+        AchievementMessagePreProcessEvent preEvent = DiscordSRV.api.callEvent(new AchievementMessagePreProcessEvent(channelName, message, player, achievementName));
```

Comment:
Cancellation should be checked

Reply:
Changed this to be consistent with your other uses (consistent API). So the results of it being cancelled are passed into the post-processing event, then the checking of the message being cancelled is done after the post-processing event.

Reply:
Doesn't appear to be the case, checked all events with Pre & Post process scheme, they all would return if preprocess was cancelled

Reply:
https://github.com/DiscordSRV/DiscordSRV/blob/develop/src/main/java/github/scarsz/discordsrv/DiscordSRV.java#L963

Reply:
No, you're right.